src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -94,32 +94,21 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
             dbsApi.insertBulkBlock(blockDump=block)
             results.put({'name': name, 'success': "uploaded"})
         except HTTPError as ex:
-            # when we will have new dbs3-client and fine granularity DBS Go server
-            # error codes then we should remove if/elif/else and replace it with
-            # DBSError(ex.body), e.g.
-            # dbsError = DBSError(ex.body)
-            # reason = dbsError.getReason()
-            # srvCode = dbsError.getServverCode()
-            # results.put({'name': name, 'success': 'error', 'error': reason, 'server_code': srvCode})
-
-            # legacy code which deal with DBS python server exceptions
             dbsError = DBSError(ex.body)
             reason = dbsError.getReason()
-            if 'Block %s already exists' % name in reason:
-                # Then this is probably a duplicate
-                # Ignore this for now
+            message = dbsError.getMessage()
+            srvCode = dbsError.getServerCode()
+            msg = f'DBSError code: {srvCode} message: {message}, reason: {reason}'
+            if srvCode == 128:
+                # block already exist
                 logging.warning("Block %s already exists. Marking it as uploaded.", name)
-                logging.debug("Exception: %s", reason)
                 results.put({'name': name, 'success': "uploaded"})
-            # python server error produces Missing message, while
-            # Go based server produces unable to find dataset_id message
-            elif 'Missing data when inserting to dataset_parents' in reason or \
-                'unable to find dataset_id' in reason:
-                msg = "Parent dataset is not inserted yet for block %s." % name
-                logging.warning(msg)
+            elif srvCode in [132, 133, 134, 135, 136, 137, 138, 139, 140]:
+                # racing conditions
+                logging.warning("Hit a transient data race condition injecting block %s", name)
```

Comment:
Hm, given that many errors might fall into this category, maybe we should make a log record for the DBSError content as well (thus, including the `msg` to this log)

Reply:
done